lib/luanetfilter.c

```diff
@@ -0,0 +1,168 @@
+/*
+* Copyright (c) 2024 ring-0 Ltda.
+*
+* Permission is hereby granted, free of charge, to any person obtaining
+* a copy of this software and associated documentation files (the
+* "Software"), to deal in the Software without restriction, including
+* without limitation the rights to use, copy, modify, merge, publish,
+* distribute, sublicense, and/or sell copies of the Software, and to
+* permit persons to whom the Software is furnished to do so, subject to
+* the following conditions:
+*
+* The above copyright notice and this permission notice shall be
+* included in all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+#include "linux/printk.h"
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+#include <linux/module.h>
+#include <linux/version.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+
+#include <lunatik.h>
+#include <linux/printk.h>
+#include <linux/types.h>
+#include <linux/list.h>
+#include <net/net_namespace.h>
+#include <linux/netfilter.h>
+#include <linux/netfilter/x_tables.h>
+
+static struct luanetfilter_xtable_s {
+	struct list_head list;
+	lunatik_object_t *runtime;
+} luanetfilter_xtable_t ;
+
+static LIST_HEAD(luanetfilter_xtable_list);
+
+static void luanetfilter_release(void *private){
+}
+
+/* Handles target registration with the kernel. Pops the target table from the stack */
+static int luanetfilter_target_reg(lua_State *L){
+	pr_err("luanetfilter_target_reg\n");
+	lua_pop(L,1); /* pop target table */
+	return 0;
+}
+
+/* Handles target match with the kernel. Pops the match table from the stack */
+static int luanetfilter_match_reg(lua_State *L){
+	pr_err("luanetfilter_match_reg\n");
+	lua_pop(L,1); /* pop match table */
+	return 0;
+}
+
+static int luanetfilter_xtable(lua_State *L){
+	pr_err("luanetfilter_xtable\n");
+
+	luaL_checktype(L, 1, LUA_TTABLE); 
+
+	lua_getfield(L,-1,"match");
+	if(!lua_istable(L, -1)){
```

Comment:
can't you register only `match` or `target` for a given `xtable`? I thought it could be `nil`..

Reply:
Yeah, I wrote it for testing the flow.